src/com/cli/contact/Main.java

```diff
@@ -0,0 +1,129 @@
+                } else if (choice == 2) {
+                    System.out.println("List Contacts:");
+
+                    List<Person> allContact = contact.getContacts();
+
+                    for (Person person : allContact
+                    ) {
+                        showPerson(person);
+                    }
+                } else if (choice == 3) {
+                    showSubMenu();
+
+                    int choiseSearch = scanner.nextInt();
+
+                    if (choiseSearch == 1) {
+                        Searcher searcher = new SearchLastName();
+                        System.out.println("Introduce Name or Last Name:");
+                        String nameLast = scanner.next();
+                        List<Person> personList = searcher.meetSearch(nameLast, contact.getContacts());
+
+                        for (Person person : personList
+                        ) {
+                            showPerson(person);
+                        }
+                    } else if (choiseSearch == 2) {
+                        Searcher searcher = new SearchOnlyLetter();
+                        System.out.println("Introduce a Letter:");
+                        String keyLetter = scanner.next();
+                        List<Person> personList = searcher.meetSearch(keyLetter, contact.getContacts());
+
+                        for (Person person : personList
+                        ) {
+                            showPerson(person);
+                        }
+                    } else if (choiseSearch == 3) {
+                        Searcher searcher = new SearchPhone();
+                        System.out.println("Introduce a Phone to search:");
+                        String phoneNumber = scanner.next();
+                        List<Person> personList = searcher.meetSearch(phoneNumber, contact.getContacts());
+
+                        for (Person person : personList
+                        ) {
+                            showPerson(person);
+                        }
+                    }
+                }
+            }
+            System.out.println("Thank you! Good Bye.");
+    }
+
+    public static void showMenu(){
+        System.out.println("|   MENU SELECTION         |");
+        System.out.println("| Options:                 |");
+        System.out.println("|        1. Add Contact    |");
+        System.out.println("|        2. Show Contacts  |");
+        System.out.println("|        3. Search Contact |");
+        System.out.println("|        4. Exit           |");
+    }
+
+    public static void showSubMenu(){
+        System.out.println("| Options Search:                |");
+        System.out.println("|        1. By Name or Last Name |");
+        System.out.println("|        2. By Letter            |");
+        System.out.println("|        3. By Phone Number      |");
+        System.out.println("|        4. Exit                 |");
+    }
+
+    public static void showPerson(Person person){
```

Comment:
@anghelitalo Great that you have here the show person and not in the class person, this behavior is related to the Menu and not to the person, great work!